### PR TITLE
[IMP] sale timesheet : Add Demo data.

### PR DESCRIPTION
**Pull request by mep-odoo**
Task : https://www.odoo.com/web?#id=39573&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720

pad : https://pad.odoo.com/p/r.84436d70d5f81a7136b280dac11b3657

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by pga-odoo** [comment]
LGTM

**Comment by jem-odoo** [comment]
Landed at https://github.com/odoo/odoo/commit/257bf539d55f1a6322a060bb143dbbd0f8b124c1

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 1]
```diff
@@ -28,10 +27,415 @@
...
+        <record id="timesheet_7" model="account.analytic.line">
+            <field name="name">Check Requirements</field>
+            <field name="employee_id" ref="hr.employee_qdp"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(weekday=2,weeks=-2)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">1.00</field>
+            <field name="project_id" ref="project_support"/>
+            <field name="task_id" search="[('sale_line_id', '=', ref('sale_line_11'))]"/>
+        </record>
+        <record id="timesheet_8" model="account.analytic.line">
+            <field name="name">Analysis data</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Data Analysis

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 2]
```diff
@@ -28,10 +27,415 @@
...
+            <field name="employee_id" ref="hr.employee_qdp"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(weekday=3,weeks=-2)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">1.00</field>
+            <field name="project_id" ref="project_support"/>
+            <field name="task_id" search="[('sale_line_id', '=', ref('sale_line_11'))]"/>
+        </record>
+        <record id="timesheet_9" model="account.analytic.line">
+            <field name="name">Code Migrate</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Code Migration

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 3]
```diff
@@ -28,10 +27,415 @@
...
+        <record id="timesheet_12" model="account.analytic.line">
+            <field name="name">Quality  check</field>
+            <field name="employee_id" ref="hr.employee_qdp"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(weekday=2,weeks=-3)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">1.00</field>
+            <field name="project_id" ref="project_support"/>
+            <field name="task_id" search="[('sale_line_id', '=', ref('sale_line_11'))]"/>
+        </record>
+        <record id="timesheet_13" model="account.analytic.line">
+            <field name="name">Migrate data</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Data Migration

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 4]
```diff
@@ -28,10 +27,415 @@
...
+            <field name="employee_id" ref="hr.employee_qdp"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(weekday=3,weeks=-3)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">1.00</field>
+            <field name="project_id" ref="project_support"/>
+            <field name="task_id" search="[('sale_line_id', '=', ref('sale_line_11'))]"/>
+        </record>
+        <record id="timesheet_14" model="account.analytic.line">
+            <field name="name">Code improved</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Code Improvement

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 5]
```diff
@@ -28,10 +27,415 @@
...
+        <record id="timesheet_15" model="account.analytic.line">
+            <field name="name">Research and Development</field>
+            <field name="employee_id" ref="hr.employee_qdp"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(weekday=1,weeks=-2)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">8.00</field>
+            <field name="task_id" search="[('sale_line_id', '=', ref('sale_line_22'))]"/>
+            <field name="project_id" search="[('sale_line_id', '=', ref('sale_line_22'))]"/>
+        </record>
+        <record id="timesheet_16" model="account.analytic.line">
+            <field name="name">Quality Analyst</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Quality Analysis

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 6]
```diff
@@ -28,10 +27,415 @@
...
+        <record id="timesheet_17" model="account.analytic.line">
+            <field name="name">Delivery and maintenance</field>
+            <field name="employee_id" ref="hr.employee_qdp"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(days=0)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">8.00</field>
+            <field name="task_id" search="[('sale_line_id', '=', ref('sale_line_22'))]"/>
+            <field name="project_id" search="[('sale_line_id', '=', ref('sale_line_22'))]"/>
+        </record>
+        <record id="timesheet_18" model="account.analytic.line">
+            <field name="name">Code Improve</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Code Improvement

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 7]
```diff
@@ -28,10 +27,415 @@
...
+            <field name="employee_id" ref="hr.employee_qdp"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(weekday=4,weeks=-2)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">8.00</field>
+            <field name="task_id" search="[('sale_line_id', '=', ref('sale_line_22'))]"/>
+            <field name="project_id" search="[('sale_line_id', '=', ref('sale_line_22'))]"/>
+        </record>
+
+        <record id="timesheet_19" model="account.analytic.line">
+            <field name="name">Discussion new Group</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Knowledge Transfer

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 8]
```diff
@@ -28,10 +27,415 @@
...
+        <record id="timesheet_20" model="account.analytic.line">
+            <field name="name">Analysis Document</field>
+            <field name="employee_id" ref="hr.employee_root"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(weekday=0,weeks=-4)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">4.00</field>
+            <field name="project_id" ref="project_support"/>
+            <field name="task_id" search="[('sale_line_id', '=', ref('sale_line_21'))]"/>
+        </record>
+        <record id="timesheet_21" model="account.analytic.line">
+            <field name="name">Write specifications</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Specification writing

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 9]
```diff
@@ -28,10 +27,415 @@
...
+            <field name="employee_id" ref="hr.employee_root"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(weekday=0,weeks=-3)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">4.00</field>
+            <field name="project_id" ref="project_support"/>
+            <field name="task_id" search="[('sale_line_id', '=', ref('sale_line_21'))]"/>
+        </record>
+        <record id="timesheet_22" model="account.analytic.line">
+            <field name="name">Requirements metting with client</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Requirements meeting

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 10]
```diff
@@ -28,10 +27,415 @@
...
+        <record id="timesheet_25" model="account.analytic.line">
+            <field name="name">Internal Discussion</field>
+            <field name="employee_id" ref="hr.employee_root"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(months=-4, days=-13)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">8.00</field>
+            <field name="project_id" ref="project_support"/>
+            <field name="task_id" ref="project_task_internal"/>
+        </record>
+        <record id="timesheet_26" model="account.analytic.line">
+            <field name="name">Improvement the Details</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Details improvement

File: addons/sale_timesheet/data/sale_service_demo.xml [hunk 11]
```diff
@@ -28,10 +27,415 @@
...
+        <record id="timesheet_23" model="account.analytic.line">
+            <field name="name">Update Docs</field>
+            <field name="employee_id" ref="hr.employee_root"/>
+            <field name="date" eval="(DateTime.now() + relativedelta(months=-4, days=-10)).strftime('%Y-%m-%d')"/>
+            <field name="unit_amount">8.00</field>
+            <field name="project_id" ref="project_support"/>
+            <field name="task_id" ref="project_task_internal"/>
+        </record>
+        <record id="timesheet_24" model="account.analytic.line">
+            <field name="name">Discussion Agreement</field>
```
Comment (bst-odoo) [inline addons/sale_timesheet/data/sale_service_demo.xml]: Agreement meeting

**Comment by Yenthe666** [comment]
Why on earth was this even merged in? It looks like some guy with a one week crash course in English wrote all those terms. "Improvement the details"? Seriously guys, learn to English.

**Comment by mart-e** [comment]
@jem-odoo can you fix that?

**Comment by bst-odoo** [comment]
@Yenthe666 It was done by out fellow indian colleagues. Please be polite. We merged it too soon but it's only demo data, be sure it will be fixed really soon.

**Comment by Yenthe666** [comment]
@bst-odoo I know and I understand that it is not perfect, nobody is perfect. The amount of bad sentences and terms in this one are really high though. Plus this obviously wasn't reviewed in detail.
Don't take it offensive, was rather ment to motivate people to do better. Sorry if this felt offensive.

**Pull request closed without merging**